### ISSUE TITLE
Season Change Screen Constantly Appearing

### ISSUE THREAD

**Issue by Bettisaurus** [RustyMods/Seasonality#41]
Currently facing an issue where the season changing screen keeps appearing like as if the season is constantly changing - at top of my screen it stays as Winter '0' seconds. My brother also faced the exact same issue yesterday. Currently using version 3.4.0 - any solutions?

**Comment by Bettisaurus** [comment]
Add-on to this issue... I am also for some reason now immune to damage.

**Comment by RustyMods** [comment]
hmmm - I'll have to revisit timer.

immune is due to screen fading - i make you immune so people do not rage if you get murdered while screen fades to black

**Comment by khionu** [comment]
Can confirm this in 3.4.0, haven't tested 3.4.1

**Comment by LitkuReiska** [comment]
confirmed, same problem persists. It was before too but it got fixed. Back again.

**Comment by OxymoRonald** [comment]
Just wanted to add my 2 cents as well. Same issue on dedicated server setup. The server's "SeasonTimeData.yml" is updating, the clients time just doesn't seem to be syncing

**Comment by jp7189** [comment]
I have the same behavior as OxymoRonald.  I didn't have the problem with 3.3.11.  I run a dedicated server, and on 3.4+ every client that connects occasionally gets a timer that says 0 for the season and repeated screen fade.  Logging out and back in usually fixes the issue (no need to completely exit the game).

Also, every client that logs in starts with a full 60m timer on the current season.  So, clients disagree on season time depending on when they log in.  All clients do agree on the season itself, and all clients move to the next season together and are then in agreement on the time.  The season generally rotates before any of the clients get to 0 - so I assume they are properly rotating based on the server's countdown (haven't checked).

I am pretty certain this is a result of this function: "Changed timer to save last season change to disk on a file"

**Comment by RustyMods** [comment]
yeah, you are right. i changed recording last season change time to on file for this attempt.

3.4.4 - i started sending last season change to clients via server as well to prevent desyncing times.

is this comment on that version or on the previous ? 3.4.3

**Comment by jp7189** [comment]
That was on 3.4.3.  I've just installed 3.4.4 on server and clients - I'll test it out.

Thanks for your awesome work and quick reply.

**Comment by jp7189** [comment]
So far, so good.  I haven't seen any problems on 3.4.4.

**Comment by jp7189** [comment]
I'm afraid to say the problem remains the same on 3.4.4.  Still sometimes when I log in the timer is set to 0 and the screen repeatedly fades to black.  If I log out and back in, then the timer is set to 60 minutes.

**Comment by Neoony** [comment]
Same thing here on 3.4.6
For 1 player out of 3, the season was stuck at Summer 0 constantly fading out to black

video: https://cdn.discordapp.com/attachments/808691717767168021/1307150827936288838/sleepy.mp4?ex=6739eb53&is=673899d3&hm=b50c08f135b42a16a63048166e20a8967b17d47cb806e0576cba242ff5e50f79&

Had to uninstall the mod

**Comment by jp7189** [comment]
> Same thing here on 3.4.6 and dedicated windows server For 1 player out of 3, the season was stuck at Summer 0 constantly fading out to black
> 
> video: https://cdn.discordapp.com/attachments/808691717767168021/1307150827936288838/sleepy.mp4?ex=6739eb53&is=673899d3&hm=b50c08f135b42a16a63048166e20a8967b17d47cb806e0576cba242ff5e50f79&
> 
> He would constantly get it whether he relogged or not. Not sure at what point did it start for him He also overwrote all config files from me and it somehow didnt help.
> 
> Had to uninstall the mod

Some of us get this every time we launch the game.  The trick to fixing it is to just exit the server and not the game.  That has worked 100% of the time to fix the repeated fading issue.

After you fix the fading issue, it's still likely your season timer will be set to 0 until the server rotates the season and then everyone should be in sync.

**Comment by khionu** [comment]
>  The trick to fixing it is to just exit the server and not the game.
Sounds, then, like the side effect of an issue that needs to get fixed in Valheim natively, but in lieu of/while waiting for a fix upstream, this mod needs to fix how it handles improper disconnects.
